### PR TITLE
Fixed `num_chan` warning and propagate to remaining instances

### PR DESCRIPTION
**Pull request by DradeAW**
kwargs in `BinaryRecordingExtractor` were causing warnings to be thrown when loading from dict.

### CODE REVIEW

File: src/spikeinterface/core/binaryrecordingextractor.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def __init__(
             "sampling_frequency": sampling_frequency,
             "t_starts": t_starts,
             "num_channels": num_channels,
-            "num_chan": num_channels,  # TODO: This should be here at least till version 0.100.0
+            "num_chan": num_chan,  # TODO: This should be here at least till version 0.100.0
```
Comment (samuelgarcia) [inline src/spikeinterface/core/binaryrecordingextractor.py:124]: we should remove this line, I think.
There is no reason to dump this

File: src/spikeinterface/core/binaryfolder.py [hunk 2]
```diff
@@ -50,6 +50,9 @@ def __init__(self, folder_path):
 
         self._kwargs = dict(folder_path=str(folder_path.absolute()))
         self._bin_kwargs = d["kwargs"]
+        if "num_channels" not in self._bin_kwargs:
```
Comment (h-mayorquin) [inline src/spikeinterface/core/binaryfolder.py:53]: Should this be removed as well?

File: src/spikeinterface/core/core_tools.py [hunk 3]
```diff
@@ -160,15 +160,15 @@ def add_suffix(file_path, possible_suffix):
     return file_path
 
 
-def read_binary_recording(file, num_chan, dtype, time_axis=0, offset=0):
+def read_binary_recording(file, num_channels, dtype, time_axis=0, offset=0):
```
Comment (h-mayorquin) [inline src/spikeinterface/core/core_tools.py:163]: I don't think we are using this function anymore, maybe we can drop it? just a comment though, not important for this PR.

File: src/spikeinterface/core/binaryfolder.py (hunk 2 above)
Comment (h-mayorquin) [reply src/spikeinterface/core/binaryfolder.py:53]: The assertion under I mean.

Comment (alejoe91) [reply src/spikeinterface/core/binaryfolder.py:53]: either `num_channels` or `num_chan` should be in the dict right?

Comment (h-mayorquin) [reply src/spikeinterface/core/binaryfolder.py:53]: Shouldn't it be only `num_channels` now?

File: src/spikeinterface/core/binaryfolder.py [hunk 4]
```diff
@@ -50,6 +50,9 @@ def __init__(self, folder_path):
...
+            assert "num_chan" in self._bin_kwargs, "Cannot find num_channels or num_chan in binary.json"
+            self._bin_kwargs["num_channels"] = self._bin_kwargs["num_chan"]
```
Comment (h-mayorquin) [inline src/spikeinterface/core/binaryfolder.py:55]: Here, I think I just quoted the wrong line, @alejoe91

File: src/spikeinterface/core/binaryfolder.py (hunk 2 above)
Comment (alejoe91) [reply src/spikeinterface/core/binaryfolder.py:53]: yes, but people have been saving in binary format for a while now! So I think this is needed for back-compatibility!

Comment (DradeAW) [reply src/spikeinterface/core/binaryfolder.py:53]: Then we should add a warning that this will not be supported in SI >= 0.100?

Comment (h-mayorquin) [reply src/spikeinterface/core/binaryfolder.py:53]: Ah, fair enough, makes sense!

Comment (h-mayorquin) [reply src/spikeinterface/core/binaryfolder.py:53]: In my view a deprecation warning makes sense when loading old files makes sense.

Comment (DradeAW) [reply src/spikeinterface/core/binaryfolder.py:53]: I mean it's already implemented in the `WaveformExtractor` load

Comment (alejoe91) [reply src/spikeinterface/core/binaryfolder.py:54]: What would a deprecation warning do here? Let's say one saved a recording object with 0.97 and stored the folder ina remote drive/cloud. I don't think we need to require them to overwrite the object with a new version, but we should still be able to load it! 

If one uses a newer version of SI and passed the `num_chan` argument, (s)he will get the warning anyways!

Comment (DradeAW) [reply src/spikeinterface/core/binaryfolder.py:54]: > I don't think we need to require them to overwrite the object with a new version, but we should still be able to load it!

Isn't that what we do with the `WaveformExtractor`?

Comment (alejoe91) [reply src/spikeinterface/core/binaryfolder.py:54]: Where?

File: src/spikeinterface/core/binaryfolder.py [hunk 5]
```diff
@@ -52,6 +53,7 @@ def __init__(self, folder_path):
         self._bin_kwargs = d["kwargs"]
         if "num_channels" not in self._bin_kwargs:
             assert "num_chan" in self._bin_kwargs, "Cannot find num_channels or num_chan in binary.json"
+            warnings.warn(f"`num_chan` contained in folder {folder_path} is deprecated: it will not be redeable starting with version 0.100.")
```
Comment (alejoe91) [inline src/spikeinterface/core/binaryfolder.py]: We should maintain back-compatibility even after 0.100!

Comment (alejoe91) [reply src/spikeinterface/core/binaryfolder.py]: So @DradeAW @h-mayorquin I would remove the deprecation warning at loading time...

File: src/spikeinterface/core/binaryfolder.py (hunk 2 above)
Comment (DradeAW) [reply src/spikeinterface/core/binaryfolder.py:54]: Ah I guess it was removed. In the past when the version changed there was a warning with the previously-loaded waveform extractors :)

File: src/spikeinterface/core/binaryfolder.py (hunk 5 above)
Comment (DradeAW) [reply src/spikeinterface/core/binaryfolder.py]: Removed.

**Review by alejoe91 — APPROVED** [review]

**Pull request merged**